MIGRATING.md

```diff
@@ -0,0 +1,92 @@
+   
+- `com.okta.sdk.resource.group.GroupList` 
+   - `listGroups(String q, String filter, String expand)`
+   Signature changed to `listGroups(String q, String filter)`
+
+- `com.okta.sdk.resource.application.SwaThreeFieldApplicationSettingsApplication`
+   - `getTargetUrl` Renamed to `getTargetURL`
+   - `setTargetUrl` Renamed to `setTargetURL`
+
+- `com.okta.sdk.resource.group.Group`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.group.GroupType`
+   
+- `com.okta.sdk.resource.group.rule.GroupRule`
+   - `delete(Boolean removeUsers)` Signature changed to `delete()`
+   - Removed `getAllGroupsValid` and `setAllGroupsValid` (property `allGroupsValid` is removed)
+   - Removed `getEmbedded` (property `_embedded` is removed)
+
+- `com.okta.sdk.resource.log.LogAuthenticationContext`
+   - `getCredentialProvider` Return type changed from `List<com.okta.sdk.resource.log.LogCredentialProvider>` to `com.okta.sdk.resource.log.LogCredentialProvider`
+   - `getCredentialType` Return type changed from `List<com.okta.sdk.resource.log.LogCredentialType>` to `com.okta.sdk.resource.log.LogCredentialType`
+
+- `com.okta.sdk.resource.policy.PolicyRule`
+   - Removed `setId` (property `id` is `read-only`)
+
+- `com.okta.sdk.resource.user.factor.VerifyFactorRequest`
+   - Removed `getTokenLifetimeSeconds` & `setTokenLifetimeSeconds` (property `tokenLifetimeSeconds` is removed)
+
+- `com.okta.sdk.resource.user.Role`
+   - `getType` Return type changed from `String` to `com.okta.sdk.resource.role.RoleType`
+   - `setType` Param type changed from `String` to an enum `com.okta.sdk.resource.role.RoleType`
+
+- All `*Factor` classes Renamed to `*UserFactor`
+
+- `com.okta.sdk.resource.user.User`
+   - Renamed `addGroupTarget` to `addGroupTargetToRole`
+   - Renamed `addRole` to `assignRole`
+   - Renamed `listGroupTargetsForRole` to `listGroupTargets`
+   - Renamed `removeGroupTargetFromRole` to `removeGroupTarget`
+   - Renamed `addFactor` to `enrollFactor`
+   - Renamed `listRoles`to `listAssignedRoles`
+   - Removed `forgotPassword` (use `resetPassword` instead)
+   - Removed `endAllSessions` (use `clearSessions` instead)
+   - `listSupportedFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList`
+   - `expirePassword` Return type changed from `com.okta.sdk.resource.user.TempPassword` to `com.okta.sdk.resource.user.User`
+   - `getFactor` Return type changed from `com.okta.sdk.resource.user.factor.Factor` to `com.okta.sdk.resource.user.factor.UserFactor`
+   - `listFactors` Return type changed from `com.okta.sdk.resource.user.factor.FactorList` to `com.okta.sdk.resource.user.factor.UserFactorList` 
+   - `resetPassword(String provider, Boolean sendEmail)` Signature changed to `resetPassword(Boolean sendEmail)`
+
+- `com.okta.sdk.resource.user.UserCredentials`
+   - Removed `getEmails` & `setEmails` (property `emails` was removed)
```

Comment:
Why?

Reply:
I am not sure why this property was removed from the latest spec. 

@bretterer do you have any background on this change?

Reply:
`email` is not documented as part of the `UserCredentials` setup.

Reply:
@bdemers @bretterer what is your recommendation here? should we tell the reader that the property does not exist anymore or?

Reply:
What was the usecase of this property in the past.  What is the way to do the same thing in the new version of the SDK?

Reply:
I'm not entirely sure of the past use case of this property. I can see that _email_ can be also set as User property (setter in User model). So, I can't tell if this property in _UserCredentials_ was used effectively in the past.